### PR TITLE
[FrameworkBundle] Fix precedence of xdebug.file_link_format

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | https://github.com/symfony/symfony-docs/pull/5625 |

See https://github.com/symfony/symfony-demo/pull/148#issuecomment-131058728

### CODE REVIEW

**Comment by javiereguiluz** [comment]
:+1:

Should we also simplify the constructor of the Twig's Code extension? See https://github.com/symfony/symfony/blob/2.3/src/Symfony/Bridge/Twig/Extension/CodeExtension.php#L34

**Comment by nicolas-grekas** [comment]
@javiereguiluz the bridge could be used without the FrameworkBundle

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 1]
```diff
@@ -356,14 +356,17 @@ private function registerTemplatingConfiguration(array $config, $ide, ContainerB
...
+            $links = array(
+                'textmate' => 'txmt://open?url=file://%%f&line=%%l',
+                'macvim' => 'mvim://open?url=file://%%f&line=%%l',
+                'emacs' => 'emacs://open?url=file://%%f&line=%%l',
+                'sublime' => 'subl://open?url=file://%%f&line=%%l',
+            );
 
-        $container->setParameter('templating.helper.code.file_link_format', isset($links[$ide]) ? $links[$ide] : $ide);
+            $link = ini_get('xdebug.file_link_format') ?: get_cfg_var('xdebug.file_link_format') ?: $ide;
+            $container->setParameter('templating.helper.code.file_link_format', isset($links[$link]) ? $links[$link] : $link);
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: checking ``$links[$link]`` for values coming from XDebug config does not make sense IMO

Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: fixed

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**